test/integration/smart_answer_flows/calculate_your_holiday_entitlement_test.rb

```diff
@@ -0,0 +1,379 @@
+require_relative "../../test_helper"
+require_relative "flow_test_helper"
+
+require "smart_answer_flows/calculate-your-holiday-entitlement"
+
+class CalculateYourHolidayEntitlementTest < ActiveSupport::TestCase
+  include FlowTestHelper
+
+  setup do
+    setup_for_testing_flow SmartAnswer::CalculateYourHolidayEntitlementFlow
+  end
+
+  context "start page" do
+    should see_start_page.with_title("Calculate holiday entitlement")
+  end
+
+  context "question :basis of calculation?" do
+    should see_question(:basis_of_calculation?).with_title("Is the holiday entitlement based on:")
+
+    should "take people who work shifts to the shift_worker_basis? question" do
+      add_response "shift-worker"
+      assert_equal :shift_worker_basis?, next_node_name
+    end
+
+    should "take people who don't work shifts to the calculation_period? question" do
+      add_response "days-worked-per-week"
+      assert_equal :calculation_period?, next_node_name
+    end
+  end
+
+  context "question :calculation period?" do
+    setup do
+      responses = { basis_of_calculation?: "days-worked-per-week" }
+    end
+
+    should see_question(:calculation_period?).with_title("Do you want to work out holiday:")
+
+    should accept_answer("starting").with_next_node(:what_is_your_starting_date?)
+    should accept_answer("starting-and-leaving").with_next_node(:what_is_your_starting_date?)
+    should accept_answer("leaving").with_next_node(:what_is_your_leaving_date?)
+
+    context "when the answer is full-year" do
+      setup do
+        responses[:calculation_period?] = "full-year"
+      end
+
+      should "take people with irregular hours to the :irregular_and_annualised_done outcome" do
+        responses[:basis_of_calculation?] = "irregular-hours"
+        assert_equal :irregular_and_annualised_done, next_node
+      end
+
+      should "take people with annualised hours to the :irregular_and_annualised_done outcome" do
+        responses[:basis_of_calculation?] = "annualised-hours"
+        assert_equal :irregular_and_annualised_done, next_node
+      end
+
+      should "take people who want a days worked per week calculation to the :how_many_days_per_week? question" do
+        responses[:basis_of_calculation?] = "days-worked-per-week"
+        assert_equal :how_many_days_per_week?, next_node
+      end
+
+      should "take people who want other types of calculation to the :how_many_hours_per_week? question" do
+        responses[:basis_of_calculation?] = "compressed-hours"
+        assert_equal :how_many_hours_per_week?, next_node
+      end
+    end
+  end
+
+  context "question :how_many_days_per_week?" do
+    setup do
+      responses = {
+        basis_of_calculation?: "days-worked-per-week",
+        calculation_period?: "full-year",
+      }
+    end
+
+    should see_question(:how_many_days_per_week?).with_title("Number of days worked per week?")
+
+    should_not accept_answer(0)
+    should_not accept_answer(8)
```

Comment:
These should be in a context block to document what they mean, they otherwise seem a bit inexplicable. I think this point applies to a few places.

```suggestion
    context "when given an invalid day of the week" do
      should_not accept_answer(0)
      should_not accept_answer(8)
    end
```

Reply:
Yes, that does make them clearer - thanks.  All instances of this fixed.